### PR TITLE
Event Wizard

### LINKED ISSUE #1115: Update trusted api for 2015 game
W/L/T is currently required, but has no meaning in the 2015 game, etc.

### LINKED ISSUE #1190: Offseason "Event Wizard" for Live Data
We should definitely extend the live-input page @fangeugene wrote and make it featured enough to submit all data for an event.

Looking through the FMS user guide, here are the important steps in the real Event Wizard we should implement:
 - Event selection. Either via dropdown of current live events, or by key input
 - Team selection (either by entering a comma separated list, or uploading team list report from FMS)
 - Schedule import (by FMS report)
 - Match play. Select mode (quals or elims), populate existing matches and enter scores. Like the matchinput page used this weekend. There should be an option to auto-calculate rankings on score input, because it would be a  pain to export/upload rankings all the time. It shouldn't be too hard to import our own rankings calculations.
 - Alliances. We can probably do import by csv like we have now. Easy enough to type in. Can do by FMS report if someone can find an example of one
 - Awards. Fields for award name, team number, and person. 

We should try and keep all of these requests to be AJAX so keys don't have to be entered over and over. Looks like we can use [this](https://github.com/SheetJS/js-xlsx) to parse FMS reports client side.

### PR DESCRIPTION
**Pull request by phil-lopreiato**
Adds an "Event Wizard" using the trusted API. Can be found at `tba.com/eventwizard`. You can input your auth keys, store them locally, and input Teams Attending, Schedule, Match Play, Match Results, Alliances, and Rankings. Schedule, Match Results, and Rankings can be input via FMS reports, which are all parsed client side. 

Changed the trusted API to take W/L/T out of required parameters (moved to generic breakdown category) to accommodate this year's game. This will require an update to Cheesy Arena (which I'll take care of).

Closes #1115 and closes #1190

### CODE REVIEW

**Comment by fangeugene** [comment]
Screenshots!

**Comment by phil-lopreiato** [comment]
screenshot from 2015-05-09 13 48 12
screenshot from 2015-05-09 13 48 29
screenshot from 2015-05-09 13 48 40
screenshot from 2015-05-09 13 49 25
screenshot from 2015-05-09 13 49 36
screenshot from 2015-05-09 13 49 43
screenshot from 2015-05-09 13 49 50

**Comment by fangeugene** [comment]
Can you fix merge issues? Thanks.

**Comment by phil-lopreiato** [comment]
All set

**Comment by fangeugene** [comment]
This is awesome. A few fixes:
- Explanation of what load auth and store auth do. I'm assuming it stores the ID and Secret to a cookie, corresponding to the event key?
- Schedule Import, Match Results, and Rankings should be more consistent. FMS Schedule Import, FMS Match Import, and FMS Rankings Import, perhaps? The description under Match currently doesn't say FMS, while the other two do. Fix up the consistency.

**Comment by fangeugene** [comment]
Also, is the only event option "other" currently?

**Comment by phil-lopreiato** [comment]
When #1107 gets merged, I was going to make that show currently live events

File: main.py [hunk 1]
```diff
@@ -61,6 +62,7 @@ def __call__(self, request, response, exception):
       RedirectRoute(r'/events/<district_abbrev:[a-z]+>/<year:[0-9]+>', DistrictDetail, 'district-detail', strict_slash=True),
       RedirectRoute(r'/events/<district_abbrev:[a-z]+>', DistrictDetail, 'district-canonical', strict_slash=True),
       RedirectRoute(r'/events', EventList, 'event-list', strict_slash=True),
+      RedirectRoute(r'/eventwizard/', EventWizardHandler, 'event-wizard', strict_slash=True),
```
Comment (fangeugene) [inline main.py]: Remove trailing slash

File: templates/eventwizard.html [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+{% block title %}The Blue Alliance - Event Wizard{% endblock %}
+
+{% block meta_description %}Interface to update matches using The Blue Alliance's write API.{% endblock %}
+
+{% block content %}
+<div class="container">
+  <div class="row">
+    <div class="col-xs-12">
+      <h1 class="endheader">TBA Event Wizard</h1>
+      <p>A tool to help you archive all the data associated with an event</p>
```
Comment (fangeugene) [inline templates/eventwizard.html]: Change to "A tool to input event data. Contact (link) if you're running an offseason event and would like to use this tool."

**Comment by fangeugene** [comment]
There should be a way to add matches/match schedule besides the FMS report.

**Comment by phil-lopreiato** [comment]
> Explanation of what load auth and store auth do. I'm assuming it stores the ID and Secret to a cookie, corresponding to the event key?

That's correct, I can add an explanation somewhere 

> There should be a way to add matches/match schedule besides the FMS report.

What's the best way to do that? Have a csv input box like on the apiwrite page? Or have a way to dynamically add rows to the table?

File: static/javascript/tba_js/eventwizard.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    if(!authId || !authSecret){
+        alert("You must enter your auth id and secret.");
+        return false;
+    }
+
+    var auth = {};
+    auth['id'] = authId;
+    auth['secret'] = authSecret;
+
+    setCookie(eventKey+"_auth", JSON.stringify(auth));
```
Comment (fangeugene) [inline static/javascript/tba_js/eventwizard.js]: This is so totally not secure.

**Comment by fangeugene** [comment]
> What's the best way to do that? Have a csv input box like on the apiwrite page? Or have a way to dynamically add rows to the table? 

I can see instances where both are useful. I guess it's okay the way it is now, and we'll figure out a way to add it later.

Also, we should get rid of /apiwrite and /matchinput once we merge this.

File: static/javascript/tba_js/eventwizard.js (hunk 3 above)
Comment (phil-lopreiato) [inline static/javascript/tba_js/eventwizard.js]: Yup. At least it stays client side, though. Got a better idea?

**Comment by phil-lopreiato** [comment]
Oh - the other thing I wanted to do was to tweak the trusted API key permissions a bit. What if we made a unique permission for each tab. That way, if we have to do a cmp-like thing again, we don't give people an easy way to change everything about an event.

**Comment by fangeugene** [comment]
That should be easy. Right now we have two kinds of permissions, event data and match videos. We can split event data into sub permissions. We should also add a check permission endpoint, and disable tabs that don't have permission.

File: static/javascript/tba_js/eventwizard_apiwrite.js [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    error: function(data) {
+      feedback.css('background-color', '#c12e2a');
+      alert(data.responseText);
+    }
+  });
+}
+
+$('#teams-ok').click(function(){
+    if(!$("#team_list").val()){
+        alert("Please team data.");
```
Comment (fangeugene) [inline static/javascript/tba_js/eventwizard_apiwrite.js]: I think you a word here.

**Pull request merged**